<launch>

    <!-- input args -->
    <arg name="connect_rf_client" default='false'/>

    <!-- robofleet client group -->
    <group if="$(arg connect_rf_client)" ns="bag_utility_rf_client"> 

        <!--launch robofleet with desired config file-->
        <arg name="rf_cfg" default="$(env HRI_CACTI_XR_PATH)/robofleet/hl-ss-config.yaml"/>
	    <node
	    	name    = "robofleet_client"
	    	pkg     = "robofleet_client"
	    	type    = "client"
	    	output  = "screen"
	    	args	= "$(arg rf_cfg)"
	    />
    </group>

    <group ns="bag_utility">
        <!-- Load parameters from YAML file -->
        <rosparam file="$(find bag_utility)/params/params.yaml" command="load"/>

        <!-- user inputs -->
        <arg name="path" default="$(find bag_utility)/bags"/>
        <arg name="name" default="bag"/>

        <!-- set params -->
        <param name="output_directory" value="$(arg path)" />
        <param name="postfix" value="$(arg name)" />

        <!--launch save_bags node -->
        <node name="save_bags" pkg="bag_utility" type="save_bags" output="screen"/>
    </group>

</launch>